src/components/data-table/ha-data-table.ts

```diff
@@ -708,16 +708,29 @@ export class HaDataTable extends LitElement {
         let items = [...data];
 
         if (groupColumn) {
+          const isGroupSortColumn = this.sortColumn === this.groupColumn;
```

Comment:
Shouldn't be using `this.sortColumn` and `this.groupColumn` here as they will skip the memoization. These should be passed to `_groupData` as arguments

Reply:
The hidden column is only there for searching. Sorting should be done only on the visible columns so we don't confuse users